packages/permissions/src/utils/has-permissions.ts

```diff
@@ -245,15 +244,22 @@ const getHasDemandedDataFence = (
 
 const getDataFenceByTypeAndGroup = (
   actualDataFences: TDataFences | null,
-  demandedDataFenceType: TDataFenceType,
+  demandedDataFenceType: string,
   demandedDataFenceGroup: string
 ) => {
   if (!actualDataFences) return null;
 
   if (demandedDataFenceType in actualDataFences) {
-    const actualDataFence = actualDataFences[demandedDataFenceType];
-    if (actualDataFence && demandedDataFenceGroup in actualDataFence) {
-      return actualDataFence[demandedDataFenceGroup];
+    switch (demandedDataFenceType) {
+      case 'store': {
```

Comment:
Just a consequence of making the `type` field a generic `string`.

Reply:
Can you say way you had to do that?

Reply:
👋